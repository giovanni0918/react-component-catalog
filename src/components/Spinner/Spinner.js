import * as React from "react";
import styles from "./Spinner.css";

export type SpinnerProps = { status: "Loading" | "Success" | "Failed" }

export const Spinner = (props: SpinnerProps): React.Node => {
    return props.status === "Loading" ? (
        <div className="Spinner-wrapper">
            <div className="Spinner"></div>
        </div>
    ) : <React.Fragment />;
}