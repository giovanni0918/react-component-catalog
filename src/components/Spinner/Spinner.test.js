import React from 'react';
import renderer from 'react-test-renderer';

import { Spinner } from './Spinner';

const SampleSpinner1 = (
    <Spinner status={"Loading"} />
);

const SampleSpinner2 = (
    <Spinner status={"Success"} />
);

test('It should render a div with className of Spinner-wrapper', () => {
    const testRenderer = renderer.create(<Spinner status={"Loading"} />);
    const tree = testRenderer.toJSON();
    const testInstance = testRenderer.root;
    expect(tree.props).toHaveProperty("className");
    expect(testInstance.children.length).toBe(1);
    expect(tree).toMatchSnapshot();
});

test('It should not render anything', () => {
    const testRenderer = renderer.create(SampleSpinner2);
    const tree = testRenderer.toJSON();
    const testInstance = testRenderer.root;
    expect(testInstance.children.length).toBe(0);
    expect(tree).toMatchSnapshot();
});
