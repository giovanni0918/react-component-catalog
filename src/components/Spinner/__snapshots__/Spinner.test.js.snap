// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should not render anything 1`] = `null`;

exports[`It should render a div with className of Spinner-wrapper 1`] = `
<div
  className="Spinner-wrapper"
>
  <div
    className="Spinner"
  />
</div>
`;
